src/acp/rule.test.ts

```diff
@@ -0,0 +1,704 @@
+
+describe("addRequiredRuleToPolicy", () => {
+  it("adds the rule in the required rules of the policy", () => {
+    const myPolicy = addRequiredRuleToPolicy(
+      mockPolicy("https://some.pod/policy-resource#policy"),
+      mockRule("https://some.pod/rule-resource#rule")
+    );
+    expect(
+      myPolicy.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
+          DataFactory.namedNode(ACP_ALL),
+          DataFactory.namedNode("https://some.pod/rule-resource#rule")
+        )
+      )
+    ).toBe(true);
+  });
+
+  it("does not remove the existing required rules", () => {
+    const mockedPolicy = mockPolicy("https://some.pod/policy-resource#policy", {
+      required: [mockRule("https://some.pod/rule-resource#another-rule")],
+    });
+    const myPolicy = addRequiredRuleToPolicy(
+      mockedPolicy,
+      mockRule("https://some.pod/rule-resource#a-rule")
+    );
+    expect(
+      myPolicy.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
+          DataFactory.namedNode(ACP_ALL),
+          DataFactory.namedNode("https://some.pod/rule-resource#another-rule")
+        )
+      )
+    ).toBe(true);
+  });
+
+  it("does not change the existing optional and forbidden rules", () => {
+    const mockedPolicy = mockPolicy("https://some.pod/policy-resource#policy", {
+      forbidden: [mockRule("https://some.pod/rule-resource#forbidden-rule")],
+      optional: [mockRule("https://some.pod/rule-resource#optional-rule")],
+    });
+    const myPolicy = addRequiredRuleToPolicy(
+      mockedPolicy,
+      mockRule("https://some.pod/rule-resource#optional-rule")
+    );
+    expect(
+      myPolicy.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
+          DataFactory.namedNode(ACP_ANY),
+          DataFactory.namedNode("https://some.pod/rule-resource#optional-rule")
+        )
+      )
+    ).toBe(true);
+    expect(
+      myPolicy.has(
+        DataFactory.quad(
+          DataFactory.namedNode("https://some.pod/policy-resource#policy"),
+          DataFactory.namedNode(ACP_NONE),
+          DataFactory.namedNode("https://some.pod/rule-resource#forbidden-rule")
+        )
+      )
+    ).toBe(true);
+  });
+
+  it("does not change the input policy", () => {
+    const myPolicy = mockPolicy("https://some.pod/policy-resource#policy");
+    addOptionalRuleToPolicy(
+      myPolicy,
+      mockRule("https://some.pod/rule-resource#rule")
+    );
+    expect(myPolicy.size).toEqual(0);
+  });
+});
+
+describe("setForbiddenRuleOnPolicy", () => {
+  it("sets the provided rules as the forbidden rules for the policy", () => {
+    const myPolicy = setForbiddenRuleOnPolicy(
+      mockPolicy("https://some.pod/policy-resource#policy"),
+      [
+        mockRule("https://some.pod/rule-resource#a-rule"),
+        mockRule("https://some.pod/rule-resource#another-rule"),
+      ]
```

Comment:
I'm not sure if we want to support setting multiple rules at the same time yet, because it might lead to API inconsistencies. Specifically, for e.g. `setInteger()` and the likes, we don't support setting multiple integers at the same time yet either, because we're not sure yet what the API will look like for lists vs. multiple triples with the same predicate.

(Also, `addForbiddenRuleToPolicy` also doesn't support multiple Rules at the same time, I think?)

Reply:
Right, I figured that there would be more use cases involving setting multiple rules at once compared to setting e.g. integers, but I didn't think about the issue of API consistency when it comes to lists. I'll fix this.

Reply:
Fixed in 802bbdf0a94e9fbcb7e41840d2001d499a94add7